packages/diagrams/frontend/sirius-components-diagrams/src/representation/DiagramRepresentation.d.ts

```diff
@@ -0,0 +1,21 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+/// <reference types="react" />
+import { RepresentationComponentProps } from '@eclipse-sirius/sirius-components-core';
+export declare const DiagramRepresentation: ({
+  editingContextId,
+  representationId,
+  readOnly,
+  selection,
+  setSelection,
```

Comment:
For my personal understanding, why is it needed here to reproduce the names of the fields already declared in `RepresentationComponentProps`? To indicate a specific sub-set that is needed/used in this context? (in this case it includes all fields)

Reply:
I think it helps but I'm only reproducing all of them because I've noticed that `tsc` reproduces the fields actually used on other examples.